### PR TITLE
Remove obsolete future reserved keywords from javascript.rb.

### PR DESCRIPTION
**Pull request by Masa-Shin**
These keywords are no longer reserved(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Future_reserved_keywords). There might be no backward compatibility issue because they have never really been keyword.

### CODE REVIEW

**Comment by pyrmont** [comment]
I ran it locally and tests passed fine.

I can kind of see the backwards compatibility argument but that's true about any update to a lexer that changes behaviour based on changes to the language.

**Comment by Masa-Shin** [comment]
@pyrmont Yes, would be more consistent. I'll make a PR.

**Comment by pyrmont** [comment]
Thanks! Feel free just to update this PR :)

**Comment by Masa-Shin** [comment]
Update the keywords, the declarations, and the reserved words according to the latest ES Specification.
[ECMA2018](https://www.ecma-international.org/ecma-262/9.0/index.html#sec-keywords)

Some words like 'async', 'from', and 'of' are not listed as keyword in the specification but I added them to our list of keywords because they have already been, and widely recognized as keywords. For more detail please see the commit messages.

**Comment by pyrmont** [comment]
In case it's of assistance for future reference, the following terms have special meaning in JavaScript even if they aren't formally listed as keywords or reserved words:

- [`from`](https://www.ecma-international.org/ecma-262/6.0/#sec-imports)
- [`as`](https://www.ecma-international.org/ecma-262/6.0/#sec-imports)
- [`of`](https://www.ecma-international.org/ecma-262/6.0/#sec-for-in-and-for-of-statements)
- [`async`](https://www.ecma-international.org/ecma-262/8.0/#sec-async-function-definitions)

They remain in Rouge's list for that reason.

File: lib/rouge/lexers/javascript.rb [hunk 1]
```diff
@@ -98,26 +98,23 @@ def self.detect?(text)
 
       def self.keywords
         @keywords ||= Set.new %w(
-          for in of while do break return continue switch case default
-          if else throw try catch finally new delete typeof instanceof
-          void this yield import export from as async super this
+          as async await break case catch continue debugger default delete
+          do else export finally from for if import ininstanceof new of
```
Comment (pyrmont) [inline lib/rouge/lexers/javascript.rb]: There's a typo here: `in` and `instanceof` should have a space between them.

Comment (Masa-Shin) [reply lib/rouge/lexers/javascript.rb]: Thank you for pointing that out. fixed it.

**Pull request merged**